modules/admin-ui-frontend/app/scripts/shared/directives/passwordStrengthDirective.js

```diff
@@ -0,0 +1,140 @@
+              '3rjs1la7qe', '555555', '654321', '666666', '7777777', '888888',
+              '987654321', 'aa12345678', 'abc123', 'admin', 'dragon', 'Dragon', 'google',
+              'iloveyou', 'Iloveyou', 'lovely', 'Monkey', 'mynoob', 'password',
+              'password1', 'password12', 'password123', 'princess', 'qwerty', 'qwerty123', 'qwertyuiop',
+              'Qwertyuiop', 'welcome', 'zxcvbnm', 'opencast' ];
+
+            var bar_color = 'background:white;';
+            var bar_width = 'width:0%;';
+            var bar_text = '';
+            var strength = calcStrength(scope.rule, pw);
+
+            setProgBar(strength);
+
+            function requirements(rule) {
+              return rule ? 1 : 0;
+            }
+
+            function calcStrength(rule, pw){
+
+              if (bad_passwords.indexOf(pw) > -1) {
+                strength = 0;
+                return strength;
+              }
+              // for every unused rule/requirement = -5 points
+              var usedRules = (rule - 5) * 5;
+
+              var uniqueChars = (pw.split('').sort().join('').replace(/(.)\1+/g, '').length) * 2;
+              var pw_length = pw.length * 4;
+              var lowerCase = (pw.length - pw.replace(/[a-z]/g, '').length) * 2;
+              var upperCase = (pw.length - pw.replace(/[A-Z]/g, '').length) * 2;
+              var number = (pw.length - pw.replace(/[0-9]/g, '').length) * 4;
+              var symbol = (pw.length - pw.replace(/\W/g, '').length) * 6;
+
+              var strength = usedRules + uniqueChars + pw_length + lowerCase + upperCase + number + symbol;
+              return strength;
+            }
+
+            function setProgBar(strength) {
+
+              if (strength >= 90) {
+                bar_color = 'background:#388ed6;';
+                bar_width = 'width:' + strength + '%;';
+                bar_text = 'Very strong';
+              }
+              else if (strength >= 70) {
+                bar_color = 'background:green;';
+                bar_width = 'width:' + strength + '%;';
+                bar_text = 'Strong';
+              }
+              else if (strength >= 50) {
+                bar_color = 'background:gold;';
+                bar_width = 'width:' + strength + '%;';
+                bar_text = 'Good';
+              }
+              else if (strength >= 30) {
+                bar_color = 'background:darkorange;';
+                bar_width = 'width:' + strength + '%;';
+                bar_text = 'Weak';
+              }
+              else if (strength >= 1) {
+                bar_color = 'background:red;';
+                bar_width = 'width:' + strength + '%;';
+                bar_text = 'Very weak';
+              }
+              else if (strength == 0){
+                bar_color = 'background:white;';
+                bar_width = 'width:' + 0 + '%;';
+                bar_text = 'Bad password';
+              }
+
```

Comment:
Instead of repeating this in every condition above, you could set this once here:

```suggestion
              bar_width = 'width:' + strength + '%;';
```